src/lib/y2storage/dialogs/guided_setup/select_filesystem.rb

```diff
@@ -0,0 +1,116 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may
+# find current contact information at www.novell.com.
+
+require "yast"
+require "y2storage"
+require "y2storage/dialogs/guided_setup/base"
+
+module Y2Storage
+  module Dialogs
+    class GuidedSetup
+      # Dialog to select filesystems.
+      class SelectFilesystem < Dialogs::GuidedSetup::Base
+        def root_filesystem_handler
+          btrfs = Storage::FsType_BTRFS
+          widget_update(:snapshots, widget_value(:root_filesystem) == btrfs, attr: :Enabled)
+        end
+
+        def separate_home_handler
+          widget_update(:home_filesystem, widget_value(:separate_home), attr: :Enabled)
+        end
+
+      protected
+
+        def dialog_title
+          _("Filesystem Options")
+        end
+
+        def dialog_content
+          HSquash(
+            VBox(
+              root_filesystem_widget,
+              VSpacing(2),
+              home_filesystem_widget
+            )
+          )
+        end
+
+        def root_filesystem_widget
+          VBox(
+            Left(
+              ComboBox(
+                Id(:root_filesystem), Opt(:notify), _("File System for Root Partition"),
+                [
+                  Item(Id(Storage::FsType_BTRFS), "BtrFS"),
+                  Item(Id(Storage::FsType_EXT4), "Ext4"),
+                  Item(Id(Storage::FsType_XFS), "XFS")
+                ]
```

Comment:
This list should not be a responsibility of this dialog. What about adding `Y2Storage::Filesystems::Type#suitable_for_root?` or something similar?